src/appleseed/foundation/utility/xmlelement.h

```diff
@@ -156,13 +156,14 @@ inline void XMLElement::write(const ContentType content_type)
     assert(!m_is_open);
 
     // Open the tag.
-    std::fprintf(m_file, "%s<%s", m_indenter.c_str(), m_name.c_str());
+    std::fprintf(m_file, "%s<%s", m_indenter.c_str(), replace_special_xml_characters(m_name).c_str());
 
     // Emit the attributes.
     for (const_each<AttributeVector> i = m_attributes; i; ++i)
     {
+        const std::string attribute_key = replace_special_xml_characters(i->first);
```

Comment:
Similarly, it's not possible for an attribute name to contain special characters. This should be reverted.